### ISSUE TITLE
[Bug] esm-import of @felte/common fails when using @sveltekit/adapter-node

### ISSUE THREAD

**Issue by benbender** [pablo-abc/felte#45]
Just noticed that there is a bug with `@felte/common` when building a sveltekit-app using `@sveltejs/adapter-node@next`. The problem disappears if I remove the adapter. The problem doesn't exist when doing a named import from f.e. the `felte`-package itself... 

I'm unsure why this happens as the correct esm-package seems to be built and is referenced in `@felte/common`'s package.json... There seem to be a subtle difference in the rollup-config which makes the adapter unhappy ;)

```
vite v2.4.4 building SSR bundle for production...
✓ 74 modules transformed.
.svelte-kit/output/server/app.js   199.25kb

Run npm run preview to preview your production build locally.

> Using @sveltejs/adapter-node
> Named export 'getPath' not found. The requested module '@felte/common' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from '@felte/common';
const { getPath } = pkg;

file:///[..]/.svelte-kit/output/server/app.js:2
import { getPath } from "@felte/common";
         ^^^^^^^
SyntaxError: Named export 'getPath' not found. The requested module '@felte/common' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from '@felte/common';
const { getPath } = pkg;
```

**Reproduction**

1. `npm init svelte@next felte-esm-repro && cd felte-esm-repro && npm i && npm add -D @felte/common @sveltejs/adapter-node@next`
2. Use the default for every question by hitting enter.
3. Add the following to your `svelte.config.js`:
```js
import node from '@sveltejs/adapter-node';

/** @type {import('@sveltejs/kit').Config} */
const config = {
	kit: {
  	adapter: node(),
		// hydrate the <div id="svelte"> element in src/app.html
		target: '#svelte'
	}
};

export default config;
```
4.  Add the following to `src/routes/index.svelte`:
```js
<script>
	import { getPath } from "@felte/common";
	import Counter from '$lib/Counter.svelte';

	console.log(getPath);
</script>
```

5. run `npm run build` and see it fail.

**Comment by pablo-abc** [comment]
This is really weird. The rollup configuration does change a lot between the `common` package and the rest. Mainly because the ESM build does need to keep each function as a separate module to facilitate tree-shaking for rollup. I've checked the package.json file and it does set the `module` and `exports` field appropriately.

Personally I've only used the static adapter for https://felte.dev, so now I'm wondering if `reporter-svelte` works with the node adapter since it uses `@felte/common` in a similar way as you have described.

**Comment by benbender** [comment]
I think the underlying problem is described here: https://github.com/chartjs/Chart.js/issues/9436.

Sidenote: If I add `"type": "module"` to the `package.json` of `@svelte/core`, the problem disappears, but I'm unsure if this breaks pure commonjs-setups...

**Comment by pablo-abc** [comment]
Ah, generally I think this is what's generally solved by adding the dep as a dev dep to Vite or Kit, or using the `noExternal` option of vite in either. The world does seem to be moving to full ESM packages (completely ditching CommonJS) so this is something I have been considering. I haven't had the time to test the implications if, for example, you're using this in Sapper with SSR (since it uses Rollup, not Vite). But I'd assume since Rollup also does favor ESM it shouldn't be an issue.

For reference (to anyone who might find this thread), here's [a wonderful guide to ESM packages](https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c) by Sindresorhus that I'll be using as a guide for transitioning.

**Comment by softgripper** [comment]
I'm having the same occur with the vercel adapter in my typescript project :\

**Comment by pablo-abc** [comment]
@softgripper out of curiosity. I'm guessing you're importing the `@felte/common` package directly? Or is this happening when importing `felte`?

**Comment by softgripper** [comment]
I'm using both.

I use felte, and also rolled my own typescript svelte reporter and a different implementation of the validation message.  I imported types and _get from felte.

**Comment by chloe-the-catgirl** [comment]
It's still happening with the v1 version of felte and the static adapter, at least for the dev mode. Repository for repro can be found here: https://github.com/chloe-the-catgirl/felte-issue-45-repro (`pnpm install && pnpm run dev`)

Since the `vite.ssr` property is not longer in newer versions of SvelteKit (and adding it nonetheless has no effect), this issue seems still up to date. Although there's an error on the command line, the code seems to work, at least in half of the cases (it's really flaky).

> exports is not defined in ES module scope
> This file is being treated as an ES module because it has a '.js' file extension and '/home/jasmin/repos/felte-esm-repro/node_modules/.pnpm/felte@1.0.0-next.25_svelte@3.46.4/node_modules/felte/package.json' contains "type": "module". To treat it as a CommonJS script, rename it to use the '.cjs' file extension.

**Comment by pablo-abc** [comment]
@chloe-the-catgirl weirdly enough, this was not happening in our site (although it was a few versions behind with Kit). But I managed to reproduce it with your repo. Thanks a lot for that!

Vite seems to be ignoring the `"exports"` field of our package.json somewhere when creating the server side output. I'm not sure _why_ this is happening since I'm not that familiar with Vite, but I fixed this (or at least it's not happening on my end anymore) by setting our rollup config to output the commonJS file with an extension of `.cjs`. The error should stop appearing by upgrading to `felte@1.0.0-next.26`.
